web/pages/reports.tsx

```diff
@@ -0,0 +1,297 @@
+    setModNotes((prevNotes) => ({
+      ...prevNotes,
+      [reportId]: newNote,
+    }))
+
+    await updateReport(reportId, { mod_note: newNote })
+  }
+
+  useEffect(() => {
+    const fetchOwner = async () => {
+      const ownerData: { [key: string]: any } = {}
+      for (const report of reports) {
+        const user = await getUserById(report.user_id)
+        ownerData[report.user_id] = user
+      }
+      setOwner(ownerData)
+    }
+
+    if (reports.length > 0) {
+      fetchOwner()
+    }
+  }, [reports])
+
+  if (!isAdminOrMod)
+    return (
+      <Page trackPageView={'mod reports'}>
+        <div className="mt-24 self-center">
+          You must be a Mod or Admin to view this page.
+        </div>
+      </Page>
+    )
+
+  if (initialLoading)
+    return (
+      <Page trackPageView={'mod reports'}>
+        <div className="mt-24 self-center">Loading reports...</div>
+      </Page>
+    )
+
+  const unresolvedReports = reports.filter(
+    (report) =>
+      report.status === 'new' ||
+      report.status === 'under review' ||
+      report.status === 'needs admin'
+  )
+
+  const resolvedReports = reports.filter(
+    (report) => report.status === 'resolved'
+  )
+
+  const renderContent = (content: any) => {
+    if (typeof content === 'string') {
+      return <span>{content}</span>
+    } else if (Array.isArray(content)) {
+      return content.map((item, index) => (
+        <div key={index}>{renderContent(item)}</div>
+      ))
+    } else if (typeof content === 'object') {
+      return (
+        <div>
+          {content.text && <span>{content.text}</span>}
+          {content.content &&
+            content.content.map((item: any, index: number) => (
+              <div key={index}>{renderContent(item)}</div>
+            ))}
+        </div>
+      )
+    } else {
+      return null
+    }
+  }
+
+  const renderReportList = (reportList: Report[]) => (
+    <Col className="w-full">
+      {reportList.map((report: Report) => (
+        <Row key={report.report_id} className="p-4">
+          <Col className="w-full gap-1">
+            <Row className="flex w-full items-center justify-between">
+              <div className="flex items-center">
+                {owner[report.user_id] && (
+                  <UserHovercard userId={report.user_id}>
+                    <div className="flex items-center">
+                      <Avatar
+                        username={owner[report.user_id]?.username || ''}
+                        avatarUrl={owner[report.user_id]?.avatarUrl || ''}
+                        size="sm"
+                      />
+                      <UserLink
+                        user={owner[report.user_id]}
+                        className="text-ink-800 ml-2"
+                      />
+                      {owner[report.user_id]?.isBannedFromPosting && (
+                        <BannedBadge />
+                      )}
+                    </div>
```

Comment:
I would make a component for `ReportItem` or something which has all of this and below.